### PR TITLE
Document dynamic validation/rule error messages.

### PR DESCRIPTION
**Pull request by ndm2**
Adds information on how validation and application rules can return dynamic error messages.

Needs translations.

### CODE REVIEW

File: en/core-libraries/validation.rst [hunk 1]
```diff
@@ -201,8 +203,42 @@ Then ensure that your validation method has the second context parameter. ::
     }
 
 Closures should return boolean true if the validation passes. If it fails,
-return boolean false or for a custom error message return a string.
+return boolean false or for a custom error message return a string, see the
+:ref:`Conditional/Dynamic Error Messages <dynamic_validation_error_messages`
```
Comment (markstory) [inline en/core-libraries/validation.rst]: I think you're missing a closing >

File: en/orm/validation.rst [hunk 2]
```diff
@@ -474,6 +476,46 @@ You may want to conditionally apply rules based on entity data::
         return false;
     }, 'userExists');
 
+Conditional/Dynamic Error Messages
+----------------------------------
+
+Rules, being it :ref:`custom callables <creating-a-rules-checker>`, or
```
Comment (markstory) [inline en/orm/validation.rst]: This section could be a link instead of duplicate content.

Comment (ndm2) [reply en/orm/validation.rst]: It's quite similar, but still a little bit different, rule objects vs validator objects, the rule syntax in general, the requirement to set the `errorField` option. I could imagine that some people would get things mixed up when just being referred to the validation section :worried:

Comment (markstory) [reply en/orm/validation.rst]: ok

**Pull request merged**